### ISSUE TITLE
Training performance discrepancy with Gradient Accumulation (Single 4090)

### ISSUE THREAD

**Issue by sung139** [ranrhuang/SPFSplat#14]
Hello, thank you for the open source code.

I attempted to train the model using the provided SPFSplat code on a single RTX 4090 GPU.

Due to OOM issues with the default configuration, I adjusted the settings as follows:

batch_size: 2

accumulate_grad_batches: 6

I intended to simulate an effective batch size of 12 through this configuration.

However, the evaluation results are significantly lower than those reported in the paper.

For testing, I used the checkpoint epoch_99999-step_400000.ckpt generated from this specific training run.

Could you provide any insights into why the performance drops despite preserving the effective batch size?

<img width="909" height="403" alt="Image" src="https://github.com/user-attachments/assets/bc9be433-936c-48e7-a0d4-edd98bec0b0b" />

<img width="408" height="180" alt="Image" src="https://github.com/user-attachments/assets/586c04c6-4611-48d2-a9d5-1349586edd57" />

<img width="917" height="168" alt="Image" src="https://github.com/user-attachments/assets/e70b6eb3-b24b-4eaf-b354-0059e3fdef33" />

**Comment by sung139** [comment]
This is my install dependencies:

Package                  Version      Build
------------------------ ------------ -----
accelerate               1.11.0
aiofiles                 23.2.1
aiohappyeyeballs         2.6.1
aiohttp                  3.13.2
aiosignal                1.4.0
annotated-doc            0.0.3
annotated-types          0.7.0
antlr4-python3-runtime   4.9.3
anyio                    4.11.0
attrs                    24.3.0
beartype                 0.19.0
black                    24.10.0
blosc2                   3.11.0
certifi                  2025.10.5
cffi                     1.15.1
charset-normalizer       2.1.1
click                    8.1.7
colorama                 0.4.6
colorspacious            1.1.2
contourpy                1.3.1
cycler                   0.12.1
dacite                   1.8.1
decorator                4.0.2
diff_gauss_pose          1.0.0.0
docker-pycreds           0.4.0
e3nn                     0.5.5
einops                   0.8.0
fastapi                  0.120.2
ffmpy                    0.6.4
filelock                 3.12.2
fonttools                4.55.3
frozenlist               1.5.0
fsspec                   2024.10.0
gitdb                    4.0.12
GitPython                3.1.45
gradio                   5.13.2
gradio_client            1.6.0
h11                      0.16.0
h5py                     3.15.1
hf-xet                   1.1.7
httpcore                 1.0.9
httpx                    0.28.1
huggingface-hub          0.36.0
hydra-core               1.3.2
idna                     3.4
imageio                  2.36.1
imageio-ffmpeg           0.5.1
intel-cmplr-lib-ur       2025.3.0
intel-openmp             2025.3.0
iopath                   0.1.10
jaxtyping                0.2.36
Jinja2                   3.1.2
kiwisolver               1.4.7
lazy_loader              0.4
lightning                2.4.0
lightning-utilities      0.11.9
lpips                    0.1.4
markdown-it-py           4.0.0
MarkupSafe               2.1.1
matplotlib               3.10.0
mdurl                    0.1.2
mkl                      2025.3.0
mkl_fft                  1.3.14       0
mkl_random               1.2.10       0
mkl-service              2.4.1
moviepy                  1.0.3
mpmath                   1.3.0
msgpack                  1.1.2
multidict                6.1.0
mypy-extensions          1.0.0
ndindex                  1.10.0
networkx                 3.1
numexpr                  2.14.1
numpy                    1.26.4
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.9.86
nvidia-nvtx-cu12         12.1.105
omegaconf                2.3.0
onemkl-license           2025.3.0
opencv-python-headless   4.10.0.84
opt_einsum               3.4.0
opt-einsum-fx            0.1.4
orjson                   3.11.4
packaging                24.0
pandas                   2.3.3
pathspec                 0.12.1
peft                     0.18.0
pillow                   10.4.0
pip                      25.2
platformdirs             3.10.0
plyfile                  1.1
portalocker              3.0.0
proglog                  0.1.10
propcache                0.2.1
protobuf                 5.29.1
psutil                   7.1.2
py-cpuinfo               9.0.0
pycparser                2.21
pydantic                 2.10.3
pydantic_core            2.27.1
pydub                    0.25.1
Pygments                 2.19.2
pyparsing                3.2.0
python-dateutil          2.8.2
python-multipart         0.0.20
pytorch-lightning        2.5.5
pytorch3d                0.7.8
pytz                     2025.2
PyYAML                   6.0
regex                    2025.11.3
requests                 2.31.0
rich                     14.2.0
roma                     1.5.1
ruff                     0.8.3
safehttpx                0.1.7
safetensors              0.4.5
scikit-image             0.25.0
scikit-video             1.1.11
scipy                    1.14.1
semantic-version         2.10.0
sentry-sdk               2.19.2
setproctitle             1.3.7
setuptools               80.9.0
shellingham              1.5.4
six                      1.16.0
smmap                    5.0.1
sniffio                  1.3.1
starlette                0.49.1
svg.py                   1.5.0
sympy                    1.13.1
tables                   3.10.2
tabulate                 0.9.0
tbb                      2022.3.0
tcmlib                   1.4.1
termcolor                3.2.0
tifffile                 2025.10.16
timm                     1.0.12
tokenizers               0.22.1
tomlkit                  0.13.3
torch                    2.5.1+cu121
torchaudio               2.5.1+cu121
torchmetrics             1.8.2
torchvision              0.20.1+cu121
tqdm                     4.67.1
transformers             4.57.1
trimesh                  4.9.0
triton                   3.1.0
typer                    0.20.0
typing_extensions        4.12.2
typing-inspection        0.4.1
tzdata                   2025.2
umf                      1.0.2
urllib3                  2.0.3
uvicorn                  0.38.0
wadler_lindig            0.1.7
wandb                    0.19.2
websockets               14.2
wheel                    0.45.1
xformers                 0.0.29.post1
yarl                     1.18.3

**Comment by ranrhuang** [comment]
Hi, are you also using the same RE10K dataset? Have you checked whether the issue might be caused by the dataset itself? You may also need to inspect your training curves to see if the model is actually converging. We have also released the V2 version of SPFSplat ([link](https://github.com/ranrhuang/SPFSplatV2)), which uses less GPU memory during training. You could consider trying that as well. Another option is to just set accumulate_grad_batches=1 and observe how the training curve behaves with a small batch size to inspect whether it is the problem of accumulate_grad_batches.

**Comment by Wayneee42** [comment]
Do you mind if I ask how long did you take to train a model using a single 4090 in the RE10K dataset? It will help me a lot, thanks!

**Comment by sung139** [comment]
It took about 10 days using a single 4090.
